### PR TITLE
[TFLite] Add support for passing extra parameters to the LUTPopulate transform function

### PR DESCRIPTION
**Pull request by Tessil**
Hello,

`LUTPopulate` currently takes a `float (*transform)(float)` function pointer to generate the values of the LUT. This commit provides some overloads to allow passing extra parameters to the transform function through a type-erased `void*` in the form of a `float (*transform)(float, const void*)` function pointer. A `void*` is used instead of `std::function` to keep compatibility with TFLite Micro.

It can then be used as follows for example:
```c++
LUTPopulate<int16_t>(
    input_scale, input_zp, output_scale, output_zp,
    [](float value, const void* extra_params) {
        if (value <= 0.0f) {
            const float output_scale =
                *static_cast<const float*>(extra_params);
            return std::numeric_limits<int16>::max() * output_scale;
        }
        return 1.0f / std::sqrt(value);
    },
    static_cast<const void*>(&output_scale), lut_int16);
```

This extra flexibility is useful for PRs like https://github.com/tensorflow/tensorflow/pull/51785 and other PRs we would like to raise that use the `LUTPopulate` function.

Thibaut

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @Tessil Can you please resolve conflicts? Thank you!

**Comment by Tessil** [comment]
@gbaned Thanks, I resolved the conflicts.

File: tensorflow/lite/kernels/internal/common.h [hunk 1]
```diff
@@ -460,22 +475,85 @@ LUTPopulate(FloatT input_scale, int32_t input_zero_point, FloatT output_scale,
...
+// indexing and the 7 remaining bits are used for interpolation. We thus use a
+// 513-entries LUT for int16 cases, 512 for the 9-bit indexing and 1 extra entry
+// to interpolate the last value.
+template <typename T>
+constexpr int LUTSize() {
+  static_assert(std::is_same<T, uint8_t>::value ||
+                    std::is_same<T, int8_t>::value ||
+                    std::is_same<T, int16_t>::value,
+                "Only LUTs with uint8, int8 or int16 inputs are supported.");
+  if (std::is_same<T, uint8_t>::value || std::is_same<T, int8_t>::value) {
```
Comment (alankelly) [inline tensorflow/lite/kernels/internal/common.h:552]: switch sizeof(T)?

Comment (Tessil) [reply tensorflow/lite/kernels/internal/common.h]: Changed to a ternary operator as done by https://github.com/tensorflow/tensorflow/commit/17275018c3bb2f4bf21aee368dc3b1e8bca7a488 to keep C++11 compatibility.

**Review by alankelly — APPROVED** [review]

**Pull request merged**